lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/impl/NumberedSplit.java

```diff
@@ -62,6 +62,24 @@ public String toString() {
   }
 
   @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    NumberedSplit<?> that = (NumberedSplit<?>) o;
+    return index == that.index;
+  }
+
+  @Override
+  public int hashCode() {
+    return index;
```

Comment:
And `hashCode()` must match the results of other two, so members which are considered equal must have same hash code.